Vaccine/src/main/java/dk/sdu/se/f23/InVasion/vaccine/VaccineControlSystem.java

```diff
@@ -110,5 +108,11 @@ public void processEvent(Event event, World world) {
             this.lastKnownState = stateChangeEvent.getNewState();
         }
     }
+    private double dist(Point p1, Point p2){
+        if (p1 == null || p2 == null){
```

Comment:
This check is not needed since p1 can't be null and we have already checked for null on p2